LoRaEngine/LoraKeysManagerFacade/LoRaDeviceCache.cs

```diff
@@ -36,27 +37,28 @@ public LoRaDeviceCache(ILoRaDeviceCacheStore cacheStore, string devEUI, string g
             this.cacheStore = cacheStore;
             this.devEUI = devEUI;
             this.gatewayId = gatewayId;
-            this.cacheKey = cacheKey ?? devEUI;
+            this.cacheKey = devEUI;
         }
 
-        public bool TryToLock(string lockKey = null)
+        public async Task<bool> TryToLockAsync(string lockKey = null, bool block = true)
```

Comment:
I find the naming confusing, plus we changed the default implementation to "wait until lock is obtained". Here block is true by default where in RedisStore is false.

Perhaps split in two methods:
- TryToLock or TryToLockOnce that only goes the happy path, a single attempt
- WaitForLock or EnsureHasLock that waits until the lock is obtained

Reply:
I'll update to default to true for all. The idea was to have the DeviceCache default to true, but others false (as that seemed to be the general perception of the existing callers). However, I think in general, acquiring a lock, at least in my understanding, should block and wait to get the lock.